src/Passes/Source/Types/Types.hpp

```diff
@@ -0,0 +1,15 @@
+#pragma once
```

Comment:
I suggest a much more specific name for this file, something like QatTypes.h or similar. When the system becomes large and files like Types.h are present in many directories, it becomes hard to manage and there's a high chance of including a wrong file. This typically happens when there're common include directories and project properties are inherited for multiple projects. In a codebase like Windows it was a bad problem. Ideally all file names in a system should be unique.

Reply:
I am happy to make this change, but I will not have time to do it tonight.